### PR TITLE
devilutionX: update to 1.4.1 (maintainer version)

### PR DESCRIPTION
**Pull request by bugcrazy**
Required packages to build devilutionX

#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by bugcrazy** [comment]
Two new packages are created needed to build devilutionX, they are StormLib and smpq.

File: srcpkgs/StormLib/template
Comment (abenson) [inline srcpkgs/StormLib/template:1]: The new packages should have their own commits.

File: srcpkgs/smpq/template
Comment (abenson) [inline srcpkgs/smpq/template:14]: This is the default, its not necessary.

**Comment by abenson** [comment]
`StormLib` shoudl also be split into `StormLib-devel`.

File: srcpkgs/smpq/patches/fix-smpq-compilation.patch
Comment (abenson) [inline srcpkgs/smpq/patches/fix-smpq-compilation.patch:17]: Why not just remove the lines entirely?

File: srcpkgs/smpq/patches/fix-smpq-compilation.patch
Comment (abenson) [inline srcpkgs/smpq/patches/fix-smpq-compilation.patch:39]: Same here, just remove the lines

File: srcpkgs/StormLib/template
Comment (bugcrazy) [reply srcpkgs/StormLib/template:1]: I forgot that.

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template:15]: In which line, what is it not necessary?

**Comment by bugcrazy** [comment]
> `StormLib` should be split into `StormLib-devel`.

Done!

File: srcpkgs/smpq/patches/fix-smpq-compilation.patch
Comment (bugcrazy) [reply srcpkgs/smpq/patches/fix-smpq-compilation.patch]: Using (FALSE), obtain the same result as you remove the rows. I refizes the patch by removing the lines.

File: srcpkgs/smpq/patches/fix-smpq-compilation.patch
Comment (bugcrazy) [reply srcpkgs/smpq/patches/fix-smpq-compilation.patch:41]: I removed the lines

File: srcpkgs/StormLib/template
Comment (bugcrazy) [reply srcpkgs/StormLib/template:1]: I opened PR for [Stormlib](34108)

Comment (abenson) [reply srcpkgs/StormLib/template:1]: No, the commit should still be in this PR, just needs to be a separate commit.

Comment (bugcrazy) [reply srcpkgs/StormLib/template:1]: It would not be better for separate PR, I say StormLib, smpq and devilutionX, each having your PR, devilutionX PR would only update, other PR would be new packages. Because it is easier to find failures, to compile everything at once and have strange results.

File: srcpkgs/smpq/template
Comment (abenson) [reply srcpkgs/smpq/template:15]: The `patch_args`

File: srcpkgs/StormLib/template
Comment (abenson) [reply srcpkgs/StormLib/template:1]: If those packages are required for the update, this should be in this PR.

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template]: Done!

File: srcpkgs/StormLib/template
Comment (bugcrazy) [reply srcpkgs/StormLib/template:1]: The creator of devilutionX software, says that the smpq package is optional, but it is not, needs this package to compile. 

https://github.com/diasurgical/devilutionX/blob/master/docs/building.md#if-you-want-to-build-the-devilutionxmpq-file-optional

**Comment by Chocimier** [comment]
StormLib-devel need to depend on StormLib, this should fix cross builds.

**Comment by bugcrazy** [comment]
> StormLib-devel need to depend on StormLib, this should fix cross builds.

Done!

**Comment by bugcrazy** [comment]
I'm researching how to solve the problem of compiling on arm architectures.

**Comment by bugcrazy** [comment]
I used the xlint tool, no problems in templates.
I await PR approval.

**Comment by bugcrazy** [comment]
@abenson , do you have a suggestion for improving the templates or a recommendation for merging?

**Comment by abenson** [comment]
Squash the commits into one per template.  Make sure they follow the used convention.

**Comment by bugcrazy** [comment]
@abenson, done: commit squashed.

**Comment by abenson** [comment]
> Squash the commits into one per template.

You squashed them all into a single commit.

File: srcpkgs/StormLib/template
Comment (abenson) [inline srcpkgs/StormLib/template:21]: depends="StormLib>=${version}_${revision}"

File: srcpkgs/smpq/template
Comment (abenson) [inline srcpkgs/smpq/template:7]: Why is this needed?  There's nothing I see in StormLib that would be needed on the host.

File: srcpkgs/smpq/template
Comment (abenson) [inline srcpkgs/smpq/template:16]: This does nothing, why is it here?

File: srcpkgs/StormLib/template
Comment (ahesford) [reply srcpkgs/StormLib/template:1]: This matter isn't resolved. Separate these new packages into individual commits in this PR.

File: srcpkgs/devilutionX/files/FindSDL2.cmake
Comment (ahesford) [inline srcpkgs/devilutionX/files/FindSDL2.cmake:1]: Where is this file coming from, why is it needed, and why is it not provided by upstream if it is actually needed?

File: srcpkgs/StormLib/template
Comment (bugcrazy) [reply srcpkgs/StormLib/template]: Done!

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template:7]: smpq depends on StormLib to create the MPQ file, version devillutionX 1.2.1 everything was built-in, now it's not.

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template:16]: The software is old, you have to go into the folder to build some files and then another folder to build the software, cmake doesn't do that.

File: srcpkgs/devilutionX/files/FindSDL2.cmake
Comment (bugcrazy) [reply srcpkgs/devilutionX/files/FindSDL2.cmake:1]: The file was present in devilutionX 1.2.1, in later versions removed, with the file, other cpu architectures (example: arm and ppc), can find SDL, without the file, SDL found by i686 and x86_64.

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template:7]: Found with "hostmakedepends", which "depends", by cmake.

**Comment by kruceter** [comment]
Marco, would you mind to take a look at my work on this package at kruceter#2?

My proposal includes the work made in #39184.

In addition to that all dependencies are fetched with xbps-src, just like they should be.

File: srcpkgs/smpq/template
Comment (abenson) [reply srcpkgs/smpq/template:16]: But this literally does not build it, this CDs into the directory and then exits.  It gets built in the do_install phase, which is unaffected by this.

File: srcpkgs/smpq/template
Comment (abenson) [reply srcpkgs/smpq/template:7]: If it depends on StormLib, StormLib-devel should go in makedepends; if its doing anything with the .so in hostmakedepends, then its wrong.  StormLib should be detected in depends as long as the entry in common/shlibs is correct and it links correctly.

File: srcpkgs/StormLib/template
Comment (bugcrazy) [reply srcpkgs/StormLib/template:1]: Done!

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template:7]: I changed "hostmakedepends" to "depends", this error message appears on compilation:

CMake Error at CMakeLists.txt:43 (message):
  StormLib or storm library not found


-- Configuring incomplete, errors occurred!
See also "/builddir/smpq-1.6/build/CMakeFiles/CMakeOutput.log".
=> ERROR: smpq-1.6_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${build_wrksrc}' exited with 1                                           
=> ERROR:   in do_configure() at common/build-style/cmake.sh:75

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template:16]: This method of entering the folder and leaving the folder, avoids this error message and completes the package creation process:

=> ERROR: smpq-1.6_1: PKGDESTDIR is empty and build_style != meta
=> ERROR: smpq-1.6_1: cannot continue with installation!

**Comment by bugcrazy** [comment]
> Marco, would you mind to take a look at my work on this package at [kruceter#2](https://github.com/kruceter/void-packages/pull/2)?
> 
> My proposal includes the work made in #39184.
> 
> In addition to that all dependencies are fetched with xbps-src, just like they should be.

I will see!

File: srcpkgs/smpq/template
Comment (abenson) [reply srcpkgs/smpq/template:7]: It shouldn't go in `depends`, either, that'll be automatically detected.  I removed the `hostmakedepends` line and the `do_build()` function (since it doesn't do anything).  `smpq` built fine.

Comment (abenson) [reply srcpkgs/smpq/template:7]: I can't recreate the issue to which you're referring; I removed the `hostmakedepends` line and the pointless `do_build()` and it builds just fine.

```
% cat srcpkgs/smpq/template 
# Template file for 'smpq'
pkgname=smpq
version=1.6
revision=1
build_style=cmake
configure_args="-DWITH_KDE=OFF"
makedepends="bzip2-devel zlib-devel StormLib-devel"
short_desc="StormLib MPQ archiving utility"
maintainer="MarcoAPC <marcoaureliopc@gmail.com>"
license="GPL-3.0-or-later"
homepage="https://launchpad.net/smpq"
distfiles="https://launchpad.net/smpq/trunk/${version}/+download/smpq_${version}.orig.tar.gz"
checksum=b5d2dc8a5de8629b71ee5d3612b6e84d88418b86c5cd39ba315e9eb0462f18cb
```

```
=> Registering new packages to /host/binpkgs/patch-9
index: added `smpq-1.6_1' (aarch64).
```

File: srcpkgs/devilutionX/template
Comment (ahesford) [inline srcpkgs/devilutionX/template:13]:
```suggestion
distfiles="
 https://github.com/diasurgical/devilutionX/archive/${version}.tar.gz
 https://raw.githubusercontent.com/diasurgical/devilutionX/1.2.1/CMake/FindSDL2.cmake
"
```

File: srcpkgs/devilutionX/files/FindSDL2.cmake
Comment (ahesford) [reply srcpkgs/devilutionX/files/FindSDL2.cmake:1]: In that case, let's not carry the file but fetch it from upstream. See the suggestions below.

File: srcpkgs/devilutionX/template
Comment (ahesford) [inline srcpkgs/devilutionX/template:18]: In addition to using the file fetched from upstream (which requires `skip_extraction` to avoid confusing `xbps-src`), these kinds of file manipulations belong in `post_extract` so that it is possible to inspect the complete version of the source tree after `xbps-src -I extract <pkg>`. Also, avoid using absolute paths like `$XBPS_BUILDDIR` and instead rely on the fact that all of these functions will run in `${wkrsrc}`.

```suggestion
skip_extraction="FindSDL2.cmake"

post_extract() {
	cp ${XBPS_SRCDISTDIR}/${pkgname}-${version}/FindSDL2.cmake CMake/finders
}
```

**Review by ahesford — CHANGES_REQUESTED** [review]

File: srcpkgs/devilutionX/template
Comment (abenson) [reply srcpkgs/devilutionX/template:18]: I agree; the less vendoring of files the better.

File: srcpkgs/devilutionX/files/FindSDL2.cmake
Comment (bugcrazy) [reply srcpkgs/devilutionX/files/FindSDL2.cmake]: Done!

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template]: Done!

File: srcpkgs/smpq/template
Comment (bugcrazy) [reply srcpkgs/smpq/template]: The problem was related to StormLib version 9.23, I updated to 9.24 and removed the lines from the smpq template.

Comment (bugcrazy) [reply srcpkgs/smpq/template]: Done!

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: Done!

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: Done!

File: srcpkgs/devilutionX/files/FindSDL2.cmake
Comment (kruceter) [reply srcpkgs/devilutionX/files/FindSDL2.cmake]: > other cpu architectures (example: arm and ppc), can find SDL, without the file, SDL found by i686 and x86_64.

I find this statement misleading and possibly false, since I was able to compile this package without the supplied `FindSDL2.cmake`.

Since I see this pull request is going nowhere, may I invite you over to #voidlinux on libera.chat to discuss your work on this package in more detail?

Thank you.

File: srcpkgs/devilutionX/template
Comment (classabbyamp) [inline srcpkgs/devilutionX/template:23]:
```suggestion
	cp ${XBPS_SRCDISTDIR}/${pkgname}-${version}/FindSDL2.cmake CMake/finders
```
indentation

File: srcpkgs/devilutionX/template
Comment (classabbyamp) [inline srcpkgs/devilutionX/template:16]:
```suggestion
distfiles="https://github.com/diasurgical/devilutionX/archive/${version}.tar.gz
 https://raw.githubusercontent.com/diasurgical/devilutionX/1.2.1/CMake/FindSDL2.cmake"
```
indentation

File: srcpkgs/StormLib/template
Comment (classabbyamp) [inline srcpkgs/StormLib/template:6]:
```suggestion
configure_args="-DBUILD_SHARED_LIBS=ON -DWITH_LIBTOMCRYPT=ON"
```
don't set cmake build type unless absolutely necessary, it is defined in the build style

File: srcpkgs/smpq/template
Comment (classabbyamp) [inline srcpkgs/smpq/template:11]:
```suggestion
homepage="https://launchpad.net/smpq"
changelog="https://bazaar.launchpad.net/~pali/smpq/trunk/view/head:/CHANGELOG"
```

Comment (bugcrazy) [reply srcpkgs/smpq/template:11]: Done!

File: srcpkgs/StormLib/template
Comment (bugcrazy) [reply srcpkgs/StormLib/template]: Done!

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: Done!

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: Done!

File: srcpkgs/devilutionX/template
Comment (kruceter) [inline srcpkgs/devilutionX/template:6]:
```suggestion
configure_args="-DVERSION_NUM=$version -DBUILD_TESTING=OFF -DDISABLE_ZERO_TIER=ON"
```

`BINARY_RELEASE` is removed upstream.

File: srcpkgs/devilutionX/template
Comment (kruceter) [inline srcpkgs/devilutionX/template:8]:
```suggestion
makedepends="SDL2-devel SDL2_image-devel libpng-devel bzip2-devel
 libsodium-devel fmt-devel zlib-devel asio"
```

File: srcpkgs/devilutionX/template
Comment (kruceter) [inline srcpkgs/devilutionX/template:14]: It builds perfectly without `FindSDL2.cmake`. Remove.

**Review by kruceter — CHANGES_REQUESTED** [review]

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: This option is no longer needed. There is a lot of inconstancy in the source code of devilutionX, a lot of code is removed and added, it makes compiling difficult.

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template:14]: You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated, it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories, due to the problem with find SDL2 on various cpus architectures.

https://pkgs.org/download/devilutionx-32bit
https://pkgs.org/download/devilutionx

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template:8]: Some software is downloaded by cmake and statically compiled, there is no need to add it to the template.

File: srcpkgs/devilutionX/files/FindSDL2.cmake
Comment (bugcrazy) [reply srcpkgs/devilutionX/files/FindSDL2.cmake]: If it were false, several distros would not have version 1.2.1 in their repositories yet, and NetBSD uses version 1.2.1, due to problems compiling on various cpu architectures.

PR is making progress yes, Void devs are suggesting improvements and standardizing template.

File: srcpkgs/devilutionX/template
Comment (kruceter) [reply srcpkgs/devilutionX/template:14]: > You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated

No, it does not make maintenance complicated. In fact, this is practiced here.

In fact, if you really want to download git revisions by abusing bandwidth every single time on every clean rebuild - be my guest. I am not sure whether committers will let this go off the hook easily.

> it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories

How come if it is not easy to compile after 1.2.1 when I could, Marco?

> due to the problem with find SDL2 on various cpus architectures.

Do you have any proof that SDL2 is not found "on various cpu architectures"?

How can you explain SDL2 being found here on void (be it native or cross) successfully without this cursed file in my case?

The urls provided by you are not convincing, i.e. not telling anything at all.

File: srcpkgs/devilutionX/template
Comment (kruceter) [reply srcpkgs/devilutionX/template:8]: Bundled source code is used provided the repository does not have them packaged. Your explanation is horribly wrong here.

File: srcpkgs/devilutionX/template
Comment (ahesford) [reply srcpkgs/devilutionX/template:14]: > You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated, it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories, due to the problem with find SDL2 on various cpus architectures.

If CMake is fetching submodules at build time, that is unacceptable. Void templates should explicitly fetch all components of a build so that we can verify that the contents are as expected and the process is as reproducible as possible.

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template:8]: I use the most performative packages and methods, if the author of devilutionX uses X package and indicates it for use, I will use it, even if it is static.

File: srcpkgs/devilutionX/template
Comment (bugcrazy) [reply srcpkgs/devilutionX/template:14]: > > You did it, making a complex template, downloading a lot of software "manually", which makes maintenance more complicated
> 
> No, it does not make maintenance complicated. In fact, this is practiced here.
> 
Yes, it's complicated, it's a simple game, not a gcc template version.
> If you really want to download git revisions by abusing bandwidth every single time on every clean rebuild - be my guest. I am not sure whether committers will let this go off the hook easily.
> 
I applied the suggestions from the Void devs, each suggestion came from a different dev, so I had to apply and resubmit the PR. Besides, version 1.2.1 always had static packages, this is nothing new.
> > it is not easy to compile versions after 1.2.1, several distros would not yet have version 1.2.1 in their repositories
> 
> How come it is not easy to compile after 1.2.1 when I could, Marco?
> 
From the 1.3.x series, the FindSDL2.cmake file is removed, so it only compiles for i686 and x86_64, the 1.2.x series had arm binary package for Linux in the github of devilutionX, in the 1.3.x series it was removed.

Then try compiling the 1.3.x series for non-x86 architectures, using the 1.2.1 template as a reference!
> > due to the problem with find SDL2 on various cpus architectures.
> 
> Do you have any proof that SDL2 is not found "on various cpu architectures"?
> 
Try compiling the 1.3.x series for non-x86, using the default cmake configuration from deviltutionx.
> How can you explain SDL2 being found here on void (be it native or cross) successfully without this cursed file in my case?
> 
You have disabled the option to download static software and download them manually, so there is no SDL2 check.
> The urls provided by you are not convincing, i.e. not telling anything at all.

You didn't look at the links, there are only three versions of devilutionX 1.2.1; 1.4.0 and 1.4.1.

  Why does version 1.2.1 exist in distro repositories?

Comment (bugcrazy) [reply srcpkgs/devilutionX/template:14]: > If CMake is fetching submodules at build time, that is unacceptable. Void templates should explicitly fetch all components of a build so that we can verify that the contents are as expected and the process is as reproducible as possible.

It's not a submodule! The same process exists since version 1.2.1.

Comment (ahesford) [reply srcpkgs/devilutionX/template:14]: That means we've been doing it wrong and it needs to be fixed.

Comment (bugcrazy) [reply srcpkgs/devilutionX/template:14]: > That means we've been doing it wrong and it needs to be fixed.

So in software made in GO and rust, is it necessary to create a template that downloads all the libraries manually, creating unnecessary lines in the template, just because the libraries don't come with source code?

What can be done is to add software that is in the Void repositories in "hostmakedepends" and "makedepends" or maybe create new packages that are not in the repositories, this is the only way to prevent cmake from downloading static software libraries, they are not submodules , I consider this approach unnecessary, as the template is ready.

Comment (ahesford) [reply srcpkgs/devilutionX/template:14]: Consider it unnecessary if you want, but that's what we expect.

**Comment by classabbyamp** [comment]
done in #41658

**Pull request closed without merging**
